src/libdredd/src/mutate_ast_consumer.cc

```diff
@@ -114,24 +115,46 @@ void MutateAstConsumer::HandleTranslationUnit(clang::ASTContext& context) {
   dredd_prelude << "  static uint64_t enabled_bitset["
                 << num_64_bit_words_required << "];\n";
   dredd_prelude << "  if (!initialized) {\n";
-  // TODO(https://github.com/mc-imperial/dredd/issues/55): Support the
-  //  environment variable featuring a comma-separated sequence of mutant ids.
-  dredd_prelude << "    const char* __dredd_environment_variable = "
+  dredd_prelude << "    const char* dredd_environment_variable = "
                    "std::getenv(\"DREDD_ENABLED_MUTATION\");\n";
-  dredd_prelude << "    if (__dredd_environment_variable != nullptr) {\n";
-  dredd_prelude << "      int value = atoi(__dredd_environment_variable);\n";
-  dredd_prelude << "      int local_value = value - " << initial_mutation_id
+  dredd_prelude << "    if (dredd_environment_variable != nullptr) {\n";
+  // The environment variable for mutations is set, so process the contents of
+  // this environment variable as a comma-seprated list of strings.
+  dredd_prelude << "      std::string contents(dredd_environment_variable);\n";
+  dredd_prelude << "      while (true) {\n";
+  // Find the position of the next comma.
+  dredd_prelude << "        size_t pos = contents.find(\",\");\n";
+  // The next token is either the prefix before the next comma (if there is a
+  // comma) or the whole string (if there is no comma).
+  dredd_prelude << "        std::string token = (pos == std::string::npos ? "
+                   "contents : contents.substr(0, pos));\n";
```

Comment:
It may be worth reversing the order of the comment to mention the whole string first, then the prefix before the next comma just so the order matches the conditional.

Reply:
Good suggestion - done!